test_rclcpp/CMakeLists.txt

```diff
@@ -135,6 +137,46 @@ if(AMENT_ENABLE_TESTING)
         "rclcpp")
     endif()
 
+    if(TARGET local_parameters_cpp__${middleware_impl})
+      ament_add_gtest(local_parameters_cpp__${middleware_impl} "test/local_parameters.cpp")
```

Comment:
The cpp file contains a main function and should therefore not link against gtest's main function.